### ISSUE TITLE
PNNX Cannot Handle Network with Single Linear Operator

### ISSUE THREAD

**Issue by liu-mengyang** [Tencent/ncnn#5347]
## error log

```
terminate called after throwing an instance of 'std::out_of_range'
  what():  map::at
Aborted (core dumped
```

## context

- Ubuntu 18.04
- PNNX 20240205

## how to reproduce

The minimal script

```python
import torch
import torch.nn as nn
import torch.nn.functional as F

class Model(nn.Module):
    def __init__(self):
        super(Model, self).__init__()

        self.linear_0 = nn.Linear(in_features=2048, out_features=1000, bias=False)

    def forward(self, x):
        x = self.linear_0(x)
        return x

def test():
    net = Model().half().float()
    net.eval()

    torch.manual_seed(0)
    x = torch.rand(2048)
    a = net(x)

    # export torchscript
    mod = torch.jit.trace(net, (x))
    mod.save("test_nn_Linear.pt")

    # torchscript to pnnx
    import os
    os.system("pnnx test_nn_Linear.pt inputshape=[1,2048]")

if __name__ == "__main__":
    if test():
        exit(0)
    else:
        exit(1)
```

## more

PNNX work again when modifying `inputshape=[1,2048]` to `inputshape=[2048]`.
But this is not conform to the general command of PNNX, where inputshape is started from batch-dim.

**Comment by nihui** [comment]
confirmed and reproduced
